css-easing-2/Overview.bs

```diff
@@ -118,20 +128,286 @@ The syntax for specifying an [=easing function=] is as follows:
+  If the input is outside the range provided,
+  the trajectory of the nearest two points is continued.
+
+  For example, here are the implicit values from the previous function:
+
+  <figure>
+    <img class="easing-graph" src="linear-with-double-input-example-continued.svg" width="400"
+      alt="linear(0, 0.25 75%, 1) plotted on a graph.
+        The graph has four points.
+        The first is at 0,0.
+        The second is at 0.25,0.25.
+        The third is at 0.75,0.25.
+        The forth is at 1,1.
+        The ends of the graph are extended at the angle of the nearest two lines.">
+  </figure>
+</div>
+<div class=example>
+  A typical use of ''linear()'' is to provide many points to create the illusion of a curve.
+
+  For example, here's how ''linear()'' could be used to create a reusable "bounce" easing function:
+
+  ```css
+  :root {
+    --bounce: linear(
+      /* Start to 1st bounce */
+      0, 0.063, 0.25, 0.563, 1 36.4%,
+      /* 1st to 2nd bounce */
+      0.812, 0.75, 0.813, 1 72.7%,
+      /* 2nd to 3rd bounce */
+      0.953, 0.938, 0.953, 1 90.9%,
+      /* 3rd bounce to end */
+      0.984, 1 100% 100%
+    );
+  }
+
+  .example {
+    animation-timing-function: var(--bounce);
+  }
+  ```
+
+  The definition ends `1 100% 100%` to create two final points,
+  so inputs greater than 1 always output 1.
+
+  <figure>
+    <img class="easing-graph" src="linear-bounce-example.svg" width="400"
+      alt="The graph of a rough bounce easing.">
+  </figure>
+
+  More points could be used to create a smoother result,
+  which may be needed for slower animations.
+</div>
+
+A <dfn export>linear easing function</dfn>
+is an [=easing function=]
+that interpolates linearly
+between its [=linear easing function/points=].
+
+A [=linear easing function=] has <dfn for="linear easing function">points</dfn>,
+a [=/list=] of [=linear easing points=].
+Initially a new empty [=/list=].
+
+A <dfn>linear easing point</dfn>
+is a [=/struct=]
+that has:
+
+<dl dfn-for="linear easing point">
+
+: <dfn>input</dfn>
+:: A number or null
+
+      Note: This is only null during [=create a linear easing function=].
+
+: <dfn>output</dfn>
+:: A number
+
+</dl>
+
+### Syntax ### {#linear-easing-function-syntax}
+
+A [=linear easing function=] has the following syntax:
+
+<dfn function lt="linear()">linear(<<linear-stop-list>>)</dfn>
+
+<pre class="prod">
+  <dfn>&lt;linear-stop-list></dfn> = [ <<linear-stop>> ]#
+  <dfn>&lt;linear-stop></dfn> = <<number>> && <<linear-stop-length>>?
+  <dfn>&lt;linear-stop-length></dfn> = <<percentage>>{1,2}
+</pre>
+
+''linear()'' is parsed into a [=linear easing function=]
+by calling [=create a linear easing function=],
+passing in its <<linear-stop-list>> as a [=/list=] of <<linear-stop>>s.
+
+### Parsing ### {#linear-easing-function-parsing}
+
+<section algorithm="to create a linear easing function">
+
+To <dfn>create a linear easing function</dfn>
+given a [=/list=] of <<linear-stop>>s |stopList|,
```

Comment:
As you’ve defined `<<linear-stop-list>>` earlier on, this could be shortened to:

```
given a <<linear-stop-list>> |stopList|,
```

Should the syntax of `<<linear-stop-list>>`  ever change, then this section can remain unchanged.

Reply:
Is the mapping of things like `<<linear-stop-list>>` to spec concepts documented anywhere? I felt like I had to do a bit of hand-waving around that.

Eg, with:

```
given a <<linear-stop-list>> |stopList|,
```

…where is it defined that |stopList| is an [infra list](https://infra.spec.whatwg.org/#list)?

Reply:
I felt like I was just kinda making it up with stuff like:

```
|stop|'s <<linear-stop-length>>'s first <<percentage>> as a number
```

Reply:
I see. Assumed the use of the `#` multiplier implies that it would become an iterable list. [Looking at the spec](https://drafts.csswg.org/css-values-4/#mult-comma) it’s not explicitly mentioned that it should be the case, so yeah … a bit of hand-waving it is 👋